cmd/wigToTrainingSet/wigToTrainingSet.go

```diff
@@ -0,0 +1,182 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/dna"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fasta"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/wig"
+	"log"
+	"math"
+	"math/rand"
+)
+
+func wigToTrainingSet(s Settings) {
+	rand.Seed(s.SetSeed)
+	var start, chromIndex int
+	var currName, lineToWrite string
+	var currFa fasta.Fasta
+	var currRand float64
+	var err error
+
+	trainOut := fileio.EasyCreate(s.TrainFile)
+	testOut := fileio.EasyCreate(s.TestFile)
+	validateOut := fileio.EasyCreate(s.ValidateFile)
+
+	if !s.NoHeader {
+		// write headers to all three output files
+		_, err = fmt.Fprintf(trainOut, "name\tseq\tvalue\n")
+		exception.PanicOnErr(err)
+		_, err = fmt.Fprintf(testOut, "name\tseq\tvalue\n")
+		exception.PanicOnErr(err)
+		_, err = fmt.Fprintf(validateOut, "name\tseq\tvalue\n")
+		exception.PanicOnErr(err)
+	}
+
+	if s.ValidationProp+s.TestingProp > 1 {
+		log.Fatalf("pVAlidation + pTesting should sum to less than one.")
+	}
+
+	wigs := wig.GoReadToChan(s.InWigFile)
+	genome := fasta.Read(s.InFastaFile)
+
+	for i := range wigs {
+		chromIndex = getChromIndex(genome, i.Chrom)
+		for start = 0; start < len(i.Values)-s.WindowSize; start += s.Stride {
+			if i.Values[(start+start+s.WindowSize)/2] == s.Missing {
```

Comment:
I might make a "midpoint" variable that could make the code a little quicker to read/understand